src/Read/ReadNode.js

```diff
@@ -46,6 +47,11 @@ FirestoreReadNode.prototype.main = function (msg, send, errorCb) {
   // remove existing listener before registering another
   this.unsubscribeListener()
 
+  if (disable) {
+    msg.firestore.query = referenceQuery
```

Comment:
Since `disable` change the payload when `true` (the payload won't be a result in this case), why not pass the query as the payload.

This mean you have 2 use cases :
1. You want a straight forward request, just give your query and get the result in payload
2. You want to build a custom query, enable this feature and get the query in the payload

IMHO this would remove some confusion by aligning with NodeRED philosophy (nodes should change the payload)

Reply:
Totally agree, Thanks a lot, I'll make the change